src/plugins/vis_types/timeseries/server/routes/vis.ts

```diff
@@ -33,6 +33,17 @@ export const visDataRoutes = (router: VisTypeTimeseriesRouter, framework: Framew
         });
       }
 
+      // My question here, for some reason this one is not working for TSVB
+      request.events.aborted$.subscribe(() => {
+        console.log('tsvb: abort is not working');
+      });
+
+      request.events.completed$.subscribe(() => {
+        console.log('tsvb: completed$ is working');
+      });
```

Comment:
The only difference I see it that in the non-working handler, you're subscribing to both events when in the other, you're only subscribing to `aborted$`. Could it be the thing? (in which case it would totally be a bug on our side)

FWIW, these observables are coming from here:

https://github.com/elastic/kibana/blob/338fe1a65d2031db4228c7d2ad2ce13fc75740f3/src/core/server/http/router/request.ts#L217-L228

Reply:
> The only difference I see it that in the non-working handler, you're subscribing to both observables when in the other, you're only subscribing to aborted$. Could it be the thing?

Nope, doesn't seem to change anything even if I don't subscribe to `completed` events.

Reply:
Hey! I just noticed that Node.js sets [the `aborted` event as deprecated](https://nodejs.org/docs/latest-v16.x/api/http.html#event-abort). It should still work for v16... but we should look at migrating away from it.